msim/src/sim/net/mod.rs

```diff
@@ -1197,6 +1256,9 @@ impl Endpoint {
 
 impl Drop for Endpoint {
     fn drop(&mut self) {
+        // all tcp sessions should already be deregistered.
+        assert!(self.live_tcp_ids.get_mut().unwrap().is_empty());
```

Comment:
What would be the risk of deregistering here in the drop rather than panicking? Explicit vs implicit?

Reply:
there's no risk - this is just documenting / enforcing the invariant that the sessions get deregistered earlier by the  TcpState Drop impl.